src/common/EventTrace.cc

```diff
@@ -61,71 +101,90 @@ void EventTrace::set_message_attrs(const Message *m, string& oid, string& contex
     ostringstream buf;
     buf << m->get_source() << "!" << m->get_source_addr() << "!"
         << m->get_tid() << "!" << m->get_seq() << "!" << m->get_type();;
-    context = buf.str();
+    evctx = buf.str();
   }
 }
 
-EventTrace::EventTrace(CephContext *_ctx, const char *_file, const char *_func, int _line) :
-  ctx(_ctx),
-  file(_file),
-  func(_func),
-  line(_line)
+EventTrace::EventTrace(CephContext *cct, const char *file, const char *func, int line) :
+  m_cct(cct), m_file(file), m_func(func), m_line(line)
 {
-  if (unlikely(!ctx)) 
-    return;
-  last_ts = ceph_clock_now();
-  init_tp(ctx);
-
-  lsubdout(ctx, eventtrace, LOG_LEVEL) << "ENTRY (" <<  func << ") " << file << ":" << line << dendl;
-  tracepoint(eventtrace, func_enter, file.c_str(), func.c_str(), line);
+  m_func_enter_ts = m_last_event_ts = ceph_clock_now();
+  if (unlikely(!init(m_cct))) return;
+  lsubdout(m_cct, eventtrace, LOG_LEVEL) << "ENTRY (" <<  m_func << ") " << m_file << ":" << m_line << dendl;
+  tracepoint(eventtrace, func_enter, m_file.c_str(), m_func.c_str(), m_line);
```

Comment:
How well does this work?  I would worry that the ceph_clock_now in ctor and dtor overhead might be significant, and that the m_counters map lookup would be slow.  Also, do we want to fill in the timestamps when the perf_counters option isn't enabled?